src/libtrx/game/level/common.c

```diff
@@ -72,6 +94,74 @@ static void M_ReadFace3(FACE3 *const face, VFILE *const file)
     face->enable_reflections = false;
 }
 
+static void M_ReadRoomMesh(const int32_t room_num, VFILE *const file)
+{
+    ROOM *const room = Room_Get(room_num);
+    const INJECTION_MESH_META inj_data = Inject_GetRoomMeshMeta(room_num);
```

Comment:
Having this dependency here is unfortunate – a better design would be to pass a dictionary of room number to injection mesh data via a parameter, like we do with `extra_pages` in `Level_ReadTexturePages`. But this would be way wordier.

Reply:
It's a bit of a hangover from TR1 injections, it'll likely change when we redo the format there. Potentially something passed to the module as a whole after inject in it with all the additional information we need here.